### PR TITLE
fix WebChatAdapter

### PR DESCRIPTION
**Pull request by billba**

### CODE REVIEW

File: samples/alarmbot-es6-botframework-webchat/src/webChatAdapter.js [hunk 1]
```diff
@@ -1,25 +1,46 @@
...
+                )
+            }
+        }
+    }
+
     post(activities) {
-        activitiesById[activities[0].replyToId] = activities;
-        activities.forEach(activity => Object.assign(activity, {id: activity.replyToId, from: {id: "The Bot"}}));
-        return Promise.resolve(activities);
+        let sentActivities = activities.map(activity => Object.assign({}, activity, {
```
Comment (Stevenic) [inline samples/alarmbot-es6-botframework-webchat/src/webChatAdapter.js]: >let [](start = 8, length = 3)

very minor nit but can be 'const'

File: samples/alarmbot-es6-botframework-webchat/src/webChatAdapter.js [hunk 2]
```diff
@@ -1,25 +1,46 @@
...
+            id: Date.now().toString(),
+            channelId: 'WebChat',
+            conversation: { id: 'bot' },
+            from: { id: 'bot' }
+        }));
+
+        sentActivities.forEach(activity => this.activity$.next(activity));
```
Comment (Stevenic) [inline samples/alarmbot-es6-botframework-webchat/src/webChatAdapter.js]: I should have caught this in the previous code review but you're not handling the 'delay' activity we made up.  This isn't an official activity type YET but Dan is considering adding it given that it's natively supported in channels like Kik and Line. We currently model 'delay' in the adapter by just doing:  setTimeout(() => { next-activity }, activity.value)

**Comment by Stevenic** [comment]
Nice consolidation... I should have noticed a few things in my original review. My bad

File: samples/alarmbot-es6-botframework-webchat/src/webChatAdapter.js (hunk 2 above)
Comment (billba) [reply samples/alarmbot-es6-botframework-webchat/src/webChatAdapter.js]: My preference would be to flow this into Web Chat and letting it handle it natively.

Comment (Stevenic) [reply samples/alarmbot-es6-botframework-webchat/src/webChatAdapter.js]: Sounds good...

**Pull request merged**